analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContext.cs

```diff
@@ -27,6 +29,10 @@ public class SonarAnalysisContext
 {
     private readonly RoslynAnalysisContext analysisContext;
     private readonly IEnumerable<DiagnosticDescriptor> supportedDiagnostics;
+    private readonly IGlobPatternMatcher globPatternMatcher;
+
+    protected static readonly ConditionalWeakTable<Compilation, ConcurrentDictionary<string, bool>> InclusionsCache = new();
```

Comment:
There is no need for 2 caches. We are only interested if we have to analyze the file or not. So once you evaluate both IsIncluded and IsExcluded the combined result should be put into a single cache.

Reply:
That actually makes a lot of sense 👍

Reply:
I also suggest to move the cache to `SonarAnalysisContextBase`

Reply:
I moved everything there